### PR TITLE
Bump parser

### PR DESCRIPTION
**Pull request by sungam3r**
Main changes in `CoreToVanillaConverter` and `SchemaBuilder`. I would not say that there are many changes or they are complex.

### CODE REVIEW

File: src/GraphQL.ApiTests/net6/GraphQL.approved.txt [hunk 1]
```diff
@@ -1545,18 +1547,12 @@ namespace GraphQL.Language.AST
...
         public string Name { get; }
         public GraphQL.Language.AST.NameNode NameNode { get; }
-        public GraphQL.Language.AST.OperationType OperationType { get; set; }
+        public GraphQLParser.AST.OperationType OperationType { get; set; }
         public GraphQL.Language.AST.SelectionSet SelectionSet { get; }
         public GraphQL.Language.AST.VariableDefinitions? Variables { get; set; }
         public override string ToString() { }
         public override void Visit<TState>(System.Action<GraphQL.Language.AST.INode, TState> action, TState state) { }
     }
-    public enum OperationType
```
Comment (sungam3r) [inline src/GraphQL.ApiTests/net6/GraphQL.approved.txt:1554]: Removed in favor of one from parser.

File: src/GraphQL.ApiTests/net6/GraphQL.approved.txt [hunk 2]
```diff
@@ -1931,56 +1927,17 @@ namespace GraphQL.Types
...
         public string? Description { get; set; }
         public virtual bool? Introspectable { get; }
-        public System.Collections.Generic.List<GraphQL.Types.DirectiveLocation> Locations { get; }
+        public System.Collections.Generic.List<GraphQLParser.AST.DirectiveLocation> Locations { get; }
         public string Name { get; set; }
         public bool Repeatable { get; set; }
         public override string ToString() { }
         public virtual void Validate(GraphQL.Types.AppliedDirective applied) { }
     }
-    public enum DirectiveLocation
```
Comment (sungam3r) [inline src/GraphQL.ApiTests/net6/GraphQL.approved.txt:1945]: Removed in favor of one from parser.

File: src/GraphQL.Tests/Complexity/ComplexityTestBase.cs [hunk 3]
```diff
@@ -10,7 +10,7 @@ public class ComplexityTestBase
         // For our heuristics in these tests it is assumed that each Field returns on average of two results.
         public ComplexityAnalyzer Analyzer { get; } = new ComplexityAnalyzer();
 
-        public IDocumentBuilder DocumentBuilder { get; } = new GraphQLDocumentBuilder();
+        public IDocumentBuilder DocumentBuilder { get; } = new GraphQLDocumentBuilder { MaxDepth = 1000 };
```
Comment (sungam3r) [inline src/GraphQL.Tests/Complexity/ComplexityTestBase.cs:13]: See `absurdly_huge_query` test. This one failed without `MaxDepth` increase.

File: src/GraphQL.Tests/CoreToVanillaConverterTests.cs [hunk 4]
```diff
@@ -64,10 +65,23 @@ public async Task DocumentExecuter_really_small_double_valid()
...
+                    ASTNodeKind.BooleanValue => new GraphQLBooleanValue(),
+                    _ => throw new NotSupportedException(),
+                };
+            }
+
             //note: thousand separators and/or culture-specific characters are invalid graphql literals, and will not be returned by graphql-parser
             //uppercase TRUE and FALSE are also invalid graphql input data, and will not be returned by graphql-parser
             //whitespace will not be returned by graphql-parser
-            _ = CoreToVanillaConverter.Value(new GraphQLScalarValue(kind) { Value = valueString });
+            dynamic node = BuildNode();
```
Comment (sungam3r) [inline src/GraphQL.Tests/CoreToVanillaConverterTests.cs:82]: 🤔 I may add interface in parser. Something like `IHasValueNode`. But I doubt.

File: src/GraphQL.Tests/Files/JSON/IntrospectionResult.json [hunk 5]
```diff
@@ -58,6 +58,12 @@
               "isDeprecated": false,
               "deprecationReason": null
             },
+            {
+              "name": "VARIABLE_DEFINITION",
```
Comment (sungam3r) [inline src/GraphQL.Tests/Files/JSON/IntrospectionResult.json:62]: #2854

File: src/GraphQL/Utilities/SchemaBuilder.cs [hunk 6]
```diff
@@ -59,14 +57,14 @@ public class SchemaBuilder
...
         {
-            var document = Parser.Parse(typeDefinitions, new ParserOptions { Ignore = IgnoreComments ? IgnoreOptions.IgnoreComments : IgnoreOptions.None });
+            var document = Parser.Parse(typeDefinitions, new ParserOptions { Ignore = IgnoreComments ? IgnoreOptions.Comments : IgnoreOptions.None });
             Validate(document);
             return BuildSchemaFrom(document);
         }
 
         protected virtual void Validate(GraphQLDocument document)
         {
-            var definitionsByName = document.Definitions.OfType<GraphQLTypeDefinition>().Where(def => !(def is GraphQLTypeExtensionDefinition)).ToLookup(def => def.Name!.Value);
```
Comment (sungam3r) [inline src/GraphQL/Utilities/SchemaBuilder.cs:69]: `GraphQLTypeExtensionDefinition` does not inherit from `GraphQLTypeDefinition` anymore.

File: src/GraphQL.Tests/CoreToVanillaConverterTests.cs (hunk 4 above)
Comment (Shane32) [reply src/GraphQL.Tests/CoreToVanillaConverterTests.cs:82]: If this only is an issue here in this test, it's okay...   I was thinking about this issue in my review of the parser, I think...

**Review by Shane32 — APPROVED** [review]

**Comment by sungam3r** [comment]
Yes, it happened only in tests.

**Pull request merged**